PythonAPI/setup.py

```diff
@@ -1,37 +1,47 @@
+"""To compile and install locally run "python setup.py build_ext --inplace".
+To install library to Python site-packages run "python setup.py build_ext install"
+"""
+import platform
+import numpy as np
```

Comment:
IIRC this line needs to go after `fetch_build_eggs` so that `numpy` can be installed automatically if not yet available

Reply:
We can include installation of `numpy` as a dependency to install `pycocotools` as `cython` is. WDYT @ppwwyyxx? This improves the linting and code structure.

Or should I change the position of the line back after `fetch_build_eggs`?

Reply:
`numpy` is a dependency to execute this `setup.py` file (used in extension declaration). Therefore declaring it as a dependency in `setup()` would be too late and IIRC it won't work. Putting it after `fetch_build_eggs` should work

Reply:
When building pycocotools in a new environment, it requires `cython` to be installed. Should the file be changed to build `cython` if not already installed?

Reply:
When it's missing, `dist.Distribution().fetch_build_eggs` takes care of installing cython before the real installation starts. This process works on Linux but don't know if things are the same on windows. If this cannot work maybe we'll have to tell users to manually install cython first

Reply:
> This process works on Linux but don't know if things are the same on windows. 

You are right, this works on Linux. But it does not work on Windows. We need to add description to install `cython` prior to the installation of pycocotools.